app/controllers/submissions_controller.rb

```diff
@@ -800,4 +826,22 @@ def get_all_file_data(revision, grouping, path)
   def parent_params
     params[:grouping_id].nil? ? super : [*super, :grouping_id]
   end
+
+  # Helper that extracts the URL from a url file to send to a user.
+  # If no URL is found, returns a string saying the file cannot be displayed.
+  def extract_url(file_content)
```

Comment:
This method can be simplified:

```ruby
def extract_url(file_content)
  url = file_content.strip
  is_valid_url?(url) ? url : I18n.t('submissions.cannot_display')
end
```

Reply:
I changed it slightly to 
`def extract_url(file_content) 
    tokens = file_content.strip.split(/\s/)
    tokens.length == 1 && is_valid_url?(tokens[0]) ? tokens[0] : I18n.t('submissions.cannot_display')
  end`
  since `URI.parse` turns whitespace into %20 characters